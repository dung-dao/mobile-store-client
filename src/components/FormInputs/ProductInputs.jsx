import React from 'react';
import {Col, Form, Input, InputNumber, Row, Select} from "antd";
import IF from "../common/IF";
import {numberValidate, requiredValidate, viValidate} from "../../utils/validate";
import PropTypes from "prop-types";

const ProductInputs = (props) => {
    const span = {span: 6};
    const readOnly = props.action === 'VIEW';
    return (
        <React.Fragment>
            <Row gutter={16}>
                <IF condt={props.action !== "CREATE"}>
                    <Col xs={24} md={12}>
                        <Form.Item
                            label="ID"
                            name="id"
                            labelCol={span}
                            labelAlign="left"
                        >
                            <Input readOnly={true}/>
                        </Form.Item>
                    </Col>
                </IF>
                <Col xs={24} md={12}>
                    <Form.Item
                        label="Tên sản phẩm"
                        name="name"
                        labelCol={span}
                        labelAlign="left"
                        rules={[
                            ...requiredValidate("Tên sản phẩm"),
                        ]}
                    >
                        <Input readOnly={readOnly}/>
                    </Form.Item>
                </Col>
                <Col xs={24} md={12}>
                    <Form.Item
                        label="Danh mục"
                        name="categoryId"
                        labelCol={span}
                        labelAlign="left"
                    >
                        <Select disabled={readOnly} defaultValue={1}>
                            <Select.Option value={1}>Điện thoại</Select.Option>
                            <Select.Option value={2}>Máy tính bảng</Select.Option>
                        </Select>
                    </Form.Item>
                </Col>
                <Col xs={24} md={12}>
                    <Form.Item
                        label="Tên mã"
                        name="codeName"
                        labelCol={span}
                        labelAlign="left"
                        rules={[
                            {required: true, message: 'Vui lòng nhập tên mã'},
                            {max: 255, message: 'Tên mã vượt quá độ dài cho phép'}
                        ]}
                    >
                        <Input readOnly={readOnly}/>
                    </Form.Item>
                </Col>
                <Col xs={24} md={12}>
                    <Form.Item
                        label="Hãng sản xuất"
                        name="manufactureId"
                        labelCol={span}
                        labelAlign="left"
                    >
                        <Select disabled={readOnly} defaultValue={1}>
                            <Select.Option value={1}>Samsung</Select.Option>
                            <Select.Option value={2}>Apple</Select.Option>
                        </Select>
                    </Form.Item>
                </Col>
                <Col xs={24} md={12}>
                    <Form.Item
                        label="Nước sản xuất"
                        name="madeIn"
                        labelCol={span}
                        labelAlign="left"
                        rules={[
                            ...viValidate("tên nước sản xuất"),
                            {max: 10000, message: 'Tên nước sản xuất vượt quá độ dài cho phép'}
                        ]}
                    >
                        <Input readOnly={readOnly}/>
                    </Form.Item>
                </Col>
                <Col xs={24} md={12}>
                    <Form.Item
                        label="Giá bán"
                        name="amount"
                        // labelCol={span}
                        // labelAlign="left"
                        rules={[
                            ...numberValidate("Giá bán"),
                            ...requiredValidate("Giá bán")
                        ]}
                    >
                        <InputNumber
                            readOnly={readOnly}
                            defaultValue={1000000}
                            min={0}
                            max={1000000000}
                            step={100000}
                        />
                    </Form.Item>
                </Col>
                <Col span={24}>
                    <Form.Item
                        label="Mô tả"
                        name="description"
                        // labelCol={span}
                        labelCol={{
                            span: 3
                        }}
                        wrapperCol={{
                            span: 20
                        }}
                        labelAlign="left"
                        rules={[
                            {max: 10000, message: 'Mô tả vượt quá độ dài cho phép'}
                        ]}
                    >
                        <Input.TextArea readOnly={readOnly}/>
                    </Form.Item>
                </Col>
            </Row>
        </React.Fragment>
    );
};

ProductInputs.propTypes = {
    initialValues: PropTypes.object,
    action: PropTypes.string.isRequired,
};

export default ProductInputs;
