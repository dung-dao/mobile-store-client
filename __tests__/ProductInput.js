import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import {configure, shallow} from 'enzyme';
import ProductInputs from "../src/components/FormInputs/ProductInputs";

configure({adapter: new Adapter()});

configure({adapter: new Adapter()});
describe('User input', () => {
    it('Match snapshoot create', () => {
        const tree = shallow(<ProductInputs span={4} action="CREATE"/>);
        expect(tree).toMatchSnapshot();
    });
    it('Match snapshoot view', () => {
        const tree = shallow(<ProductInputs span={4} action="VIEW"/>);
        expect(tree).toMatchSnapshot();
    });
})
