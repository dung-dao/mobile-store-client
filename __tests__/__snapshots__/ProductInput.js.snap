// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User input Match snapshoot create 1`] = `
<Fragment>
  <Row
    gutter={16}
  >
    <IF
      condt={false}
    >
      <Col
        md={12}
        xs={24}
      >
        <FormItem
          label="ID"
          labelAlign="left"
          labelCol={
            Object {
              "span": 6,
            }
          }
          name="id"
        >
          <Input
            readOnly={true}
            type="text"
          />
        </FormItem>
      </Col>
    </IF>
    <Col
      md={12}
      xs={24}
    >
      <FormItem
        label="Tên sản phẩm"
        labelAlign="left"
        labelCol={
          Object {
            "span": 6,
          }
        }
        name="name"
        rules={
          Array [
            Object {
              "message": "Tên sản phẩm không được để trống",
              "required": true,
            },
          ]
        }
      >
        <Input
          readOnly={false}
          type="text"
        />
      </FormItem>
    </Col>
    <Col
      md={12}
      xs={24}
    >
      <FormItem
        label="Danh mục"
        labelAlign="left"
        labelCol={
          Object {
            "span": 6,
          }
        }
        name="categoryId"
      >
        <Select
          bordered={true}
          choiceTransitionName="zoom"
          defaultValue={1}
          disabled={false}
          transitionName="slide-up"
        >
          <Option
            value={1}
          >
            Điện thoại
          </Option>
          <Option
            value={2}
          >
            Máy tính bảng
          </Option>
        </Select>
      </FormItem>
    </Col>
    <Col
      md={12}
      xs={24}
    >
      <FormItem
        label="Tên mã"
        labelAlign="left"
        labelCol={
          Object {
            "span": 6,
          }
        }
        name="codeName"
        rules={
          Array [
            Object {
              "message": "Vui lòng nhập tên mã",
              "required": true,
            },
            Object {
              "max": 255,
              "message": "Tên mã vượt quá độ dài cho phép",
            },
          ]
        }
      >
        <Input
          readOnly={false}
          type="text"
        />
      </FormItem>
    </Col>
    <Col
      md={12}
      xs={24}
    >
      <FormItem
        label="Hãng sản xuất"
        labelAlign="left"
        labelCol={
          Object {
            "span": 6,
          }
        }
        name="manufactureId"
      >
        <Select
          bordered={true}
          choiceTransitionName="zoom"
          defaultValue={1}
          disabled={false}
          transitionName="slide-up"
        >
          <Option
            value={1}
          >
            Samsung
          </Option>
          <Option
            value={2}
          >
            Apple
          </Option>
        </Select>
      </FormItem>
    </Col>
    <Col
      md={12}
      xs={24}
    >
      <FormItem
        label="Nước sản xuất"
        labelAlign="left"
        labelCol={
          Object {
            "span": 6,
          }
        }
        name="madeIn"
        rules={
          Array [
            Object {
              "maxlength": 10000,
              "message": "Tên nước sản xuất không hợp lệ",
              "pattern": /\\^\\[\\\\t-\\\\r A-Z_a-z\\\\xA0\\\\xC0-\\\\xC3\\\\xC8-\\\\xCA\\\\xCC\\\\xCD\\\\xD2-\\\\xD5\\\\xD9\\\\xDA\\\\xDD\\\\xE0-\\\\xE3\\\\xE8-\\\\xEA\\\\xEC\\\\xED\\\\xF2-\\\\xF5\\\\xF9\\\\xFA\\\\u0102\\\\u0103\\\\u0110\\\\u0111\\\\u0128\\\\u0129\\\\u0168\\\\u0169\\\\u01A0\\\\u01A1\\\\u01AF\\\\u01B0\\\\u1680\\\\u1EA0-\\\\u1EBD\\\\u1EC0-\\\\u1EF9\\\\u2000-\\\\u200A\\\\u2028\\\\u2029\\\\u202F\\\\u205F\\\\u3000\\\\uFEFF\\]\\+\\$/,
            },
            Object {
              "max": 10000,
              "message": "Tên nước sản xuất vượt quá độ dài cho phép",
            },
          ]
        }
      >
        <Input
          readOnly={false}
          type="text"
        />
      </FormItem>
    </Col>
    <Col
      md={12}
      xs={24}
    >
      <FormItem
        label="Giá bán"
        labelAlign="left"
        labelCol={
          Object {
            "span": 6,
          }
        }
        name="price"
        rules={
          Array [
            Object {
              "max": 11,
              "message": "Giá bán chỉ bao gồm số",
              "pattern": /\\^\\[0-9\\]\\+\\$/,
            },
            Object {
              "message": "Giá bán không được để trống",
              "required": true,
            },
          ]
        }
      >
        <ForwardRef
          defaultValue={1000000}
          max={1000000000}
          min={0}
          readOnly={false}
          step={100000}
        />
      </FormItem>
    </Col>
    <Col
      span={24}
    >
      <FormItem
        label="Mô tả"
        labelAlign="left"
        labelCol={
          Object {
            "span": 3,
          }
        }
        name="description"
        rules={
          Array [
            Object {
              "max": 10000,
              "message": "Mô tả vượt quá độ dài cho phép",
            },
          ]
        }
        wrapperCol={
          Object {
            "span": 20,
          }
        }
      >
        <TextArea
          readOnly={false}
        />
      </FormItem>
    </Col>
  </Row>
</Fragment>
`;

exports[`User input Match snapshoot view 1`] = `
<Fragment>
  <Row
    gutter={16}
  >
    <IF
      condt={true}
    >
      <Col
        md={12}
        xs={24}
      >
        <FormItem
          label="ID"
          labelAlign="left"
          labelCol={
            Object {
              "span": 6,
            }
          }
          name="id"
        >
          <Input
            readOnly={true}
            type="text"
          />
        </FormItem>
      </Col>
    </IF>
    <Col
      md={12}
      xs={24}
    >
      <FormItem
        label="Tên sản phẩm"
        labelAlign="left"
        labelCol={
          Object {
            "span": 6,
          }
        }
        name="name"
        rules={
          Array [
            Object {
              "message": "Tên sản phẩm không được để trống",
              "required": true,
            },
          ]
        }
      >
        <Input
          readOnly={true}
          type="text"
        />
      </FormItem>
    </Col>
    <Col
      md={12}
      xs={24}
    >
      <FormItem
        label="Danh mục"
        labelAlign="left"
        labelCol={
          Object {
            "span": 6,
          }
        }
        name="categoryId"
      >
        <Select
          bordered={true}
          choiceTransitionName="zoom"
          defaultValue={1}
          disabled={true}
          transitionName="slide-up"
        >
          <Option
            value={1}
          >
            Điện thoại
          </Option>
          <Option
            value={2}
          >
            Máy tính bảng
          </Option>
        </Select>
      </FormItem>
    </Col>
    <Col
      md={12}
      xs={24}
    >
      <FormItem
        label="Tên mã"
        labelAlign="left"
        labelCol={
          Object {
            "span": 6,
          }
        }
        name="codeName"
        rules={
          Array [
            Object {
              "message": "Vui lòng nhập tên mã",
              "required": true,
            },
            Object {
              "max": 255,
              "message": "Tên mã vượt quá độ dài cho phép",
            },
          ]
        }
      >
        <Input
          readOnly={true}
          type="text"
        />
      </FormItem>
    </Col>
    <Col
      md={12}
      xs={24}
    >
      <FormItem
        label="Hãng sản xuất"
        labelAlign="left"
        labelCol={
          Object {
            "span": 6,
          }
        }
        name="manufactureId"
      >
        <Select
          bordered={true}
          choiceTransitionName="zoom"
          defaultValue={1}
          disabled={true}
          transitionName="slide-up"
        >
          <Option
            value={1}
          >
            Samsung
          </Option>
          <Option
            value={2}
          >
            Apple
          </Option>
        </Select>
      </FormItem>
    </Col>
    <Col
      md={12}
      xs={24}
    >
      <FormItem
        label="Nước sản xuất"
        labelAlign="left"
        labelCol={
          Object {
            "span": 6,
          }
        }
        name="madeIn"
        rules={
          Array [
            Object {
              "maxlength": 10000,
              "message": "Tên nước sản xuất không hợp lệ",
              "pattern": /\\^\\[\\\\t-\\\\r A-Z_a-z\\\\xA0\\\\xC0-\\\\xC3\\\\xC8-\\\\xCA\\\\xCC\\\\xCD\\\\xD2-\\\\xD5\\\\xD9\\\\xDA\\\\xDD\\\\xE0-\\\\xE3\\\\xE8-\\\\xEA\\\\xEC\\\\xED\\\\xF2-\\\\xF5\\\\xF9\\\\xFA\\\\u0102\\\\u0103\\\\u0110\\\\u0111\\\\u0128\\\\u0129\\\\u0168\\\\u0169\\\\u01A0\\\\u01A1\\\\u01AF\\\\u01B0\\\\u1680\\\\u1EA0-\\\\u1EBD\\\\u1EC0-\\\\u1EF9\\\\u2000-\\\\u200A\\\\u2028\\\\u2029\\\\u202F\\\\u205F\\\\u3000\\\\uFEFF\\]\\+\\$/,
            },
            Object {
              "max": 10000,
              "message": "Tên nước sản xuất vượt quá độ dài cho phép",
            },
          ]
        }
      >
        <Input
          readOnly={true}
          type="text"
        />
      </FormItem>
    </Col>
    <Col
      md={12}
      xs={24}
    >
      <FormItem
        label="Giá bán"
        labelAlign="left"
        labelCol={
          Object {
            "span": 6,
          }
        }
        name="price"
        rules={
          Array [
            Object {
              "max": 11,
              "message": "Giá bán chỉ bao gồm số",
              "pattern": /\\^\\[0-9\\]\\+\\$/,
            },
            Object {
              "message": "Giá bán không được để trống",
              "required": true,
            },
          ]
        }
      >
        <ForwardRef
          defaultValue={1000000}
          max={1000000000}
          min={0}
          readOnly={true}
          step={100000}
        />
      </FormItem>
    </Col>
    <Col
      span={24}
    >
      <FormItem
        label="Mô tả"
        labelAlign="left"
        labelCol={
          Object {
            "span": 3,
          }
        }
        name="description"
        rules={
          Array [
            Object {
              "max": 10000,
              "message": "Mô tả vượt quá độ dài cho phép",
            },
          ]
        }
        wrapperCol={
          Object {
            "span": 20,
          }
        }
      >
        <TextArea
          readOnly={true}
        />
      </FormItem>
    </Col>
  </Row>
</Fragment>
`;
